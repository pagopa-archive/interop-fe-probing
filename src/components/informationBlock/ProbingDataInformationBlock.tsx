import { Chip, Grid, Typography } from '@mui/material'
import { InformationContainer } from '@pagopa/interop-fe-commons'
import RefreshIcon from '@mui/icons-material/Refresh'
import { useTranslation } from 'react-i18next'
import format from 'date-fns/format'
import { ProbingDataAlert } from '../probingDataAlert/ProbingDataAlert'
import { ServiceProbingData } from '../../types'
import { ButtonNaked } from '@pagopa/mui-italia'

interface IProps {
  probingData: ServiceProbingData
  reloadProbingDetails: () => void
}
const blocksInfo: Array<string> = ['probingEnabled', 'state', 'responseReceived']

export const ProbingDataInformationBlock: React.FC<IProps> = ({
  probingData,
  reloadProbingDetails,
}) => {
  const { t } = useTranslation(['detailsPage', 'general'])

  const getChipColor = (value: string) => {
    switch (value) {
      case 'ONLINE':
      case 'true':
        return 'success'
      case 'OFFLINE':
      case 'false':
        return 'error'
      case 'N/D':
        return 'warning'
    }
  }

  const getChipLabel = (value: boolean) => {
    switch (value) {
      case true:
        return 'active'
      case false:
        return 'suspended'
    }
  }

  return (
    <>
      <>
        <Grid item container direction="column">
          <Grid item textAlign={'center'}>
            <Typography variant="h5">{t('realTimeTitle', { ns: 'detailsPage' })}</Typography>
          </Grid>
          <Grid
            item
            container
            xs={6}
            px={2}
            my={2}
            justifyContent={'end'}
            sx={{ backgroundColor: '#F2F2F2' }}
          >
            <ButtonNaked
              color="primary"
              onClick={() => reloadProbingDetails()}
              size="small"
              startIcon={<RefreshIcon />}
            >
              {t('refresh', { ns: 'general' })}
            </ButtonNaked>
          </Grid>
          <Grid item container direction="column">
            {blocksInfo.map((block) => (
              <Grid item key={block}>
                <InformationContainer
                  key={block}
                  label={t(block, { ns: 'detailsPage' }).toUpperCase()}
                  content={
                    ['probingEnabled', 'state'].includes(block) ? (
                      <Chip
                        size={'small'}
                        label={
                          block === 'state'
                            ? probingData[block as keyof ServiceProbingData]
                                .toString()
                                .toLowerCase()
                            : t(getChipLabel(probingData.probingEnabled), {
                                ns: 'detailsPage',
                              })
                        }
                        color={getChipColor(
                          probingData[block as keyof ServiceProbingData].toString()
                        )}
                      />
                    ) : probingData['responseReceived'] ? (
                      format(
                        new Date(probingData['responseReceived']),
                        t('dateFormat', {
                          ns: 'general',
                        })
                      )
                    ) : (
                      ''
                    )
                  }
                />
              </Grid>
            ))}
          </Grid>
        </Grid>
        <ProbingDataAlert
          probingEnabled={probingData.probingEnabled}
          state={probingData.state}
          eserviceActive={probingData.eserviceActive}
        />
      </>
    </>
  )
}
