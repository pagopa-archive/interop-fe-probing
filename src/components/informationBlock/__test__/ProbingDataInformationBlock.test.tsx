import { describe, expect, test, vi } from 'vitest'
import { render } from '@testing-library/react'
import { ProbingDataInformationBlock } from '../ProbingDataInformationBlock'

describe('ProbingDataInformationBlock', () => {
  test('render component', () => {
    const { container } = render(
      <ProbingDataInformationBlock
        probingData={{
          probingEnabled: true,
          state: 'offline',
          responseReceived: '2023-04-27T14:55:25Z',
          eserviceActive: false,
        }}
        reloadProbingDetails={vi.fn()}
      />
    )
    expect(container).toMatchSnapshot()
  })

  test('render component with different data', () => {
    const { container } = render(
      <ProbingDataInformationBlock
        probingData={{
          probingEnabled: false,
          state: 'N/D',
          responseReceived: '2023-04-27T14:55:25Z',
          eserviceActive: false,
        }}
        reloadProbingDetails={vi.fn()}
      />
    )
    expect(container).not.toMatchSnapshot()
  })

  test('render component with responseReceived equal to null', () => {
    const { container } = render(
      <ProbingDataInformationBlock
        probingData={{
          probingEnabled: false,
          state: 'N/D',
          responseReceived: null,
          eserviceActive: false,
        }}
        reloadProbingDetails={vi.fn()}
      />
    )
    expect(container).not.toMatchSnapshot()
  })
})
