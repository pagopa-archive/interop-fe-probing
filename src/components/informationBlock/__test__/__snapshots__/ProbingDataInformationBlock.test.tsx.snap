// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProbingDataInformationBlock > render component 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-10omzie-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        realTimeTitle
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-mx38o7-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall css-14wetfy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </span>
        refresh
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              PROBINGENABLED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-li422f-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  active
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              STATE
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1fj5pxn-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  offline
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              RESPONSERECEIVED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              27/04/2023, at 16:55
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataInformationBlock > render component with different data 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-10omzie-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        realTimeTitle
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-mx38o7-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall css-14wetfy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </span>
        refresh
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              PROBINGENABLED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-lwe8th-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  suspended
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              STATE
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorWarning MuiChip-filledWarning css-uqjxnh-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  n/d
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              RESPONSERECEIVED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              27/04/2023, at 16:55
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        monitoringSystemSuspendedMessage
      </div>
    </div>
  </div>
</div>
`;

exports[`ProbingDataInformationBlock > render component with responseReceived equal to null 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item css-10omzie-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        realTimeTitle
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-6 css-mx38o7-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall MuiButton-root MuiButton-naked MuiButton-nakedPrimary MuiButton-sizeSmall MuiButton-nakedSizeSmall css-14wetfy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall css-y6rp3m-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefreshIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
            />
          </svg>
        </span>
        refresh
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column css-80wlp8-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              PROBINGENABLED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-lwe8th-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  suspended
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              STATE
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <div
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            >
              <div
                class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorWarning MuiChip-filledWarning css-uqjxnh-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                >
                  n/d
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <div
          class="MuiStack-root css-19qjufr-MuiStack-root"
        >
          <div
            class="MuiBox-root css-iuqs4u"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              RESPONSERECEIVED
            </p>
          </div>
          <div
            class="MuiBox-root css-1rr4qq7"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-hoiwr4-MuiTypography-root"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-1j916qe-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="ReportProblemOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
          />
        </svg>
      </div>
      <div
        class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        monitoringSystemSuspendedMessage
      </div>
    </div>
  </div>
</div>
`;
